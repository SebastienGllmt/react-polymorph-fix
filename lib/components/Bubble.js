'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Bubble = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _createRef = require('create-react-ref/lib/createRef');

var _createRef2 = _interopRequireDefault(_createRef);

var _withTheme = require('./HOC/withTheme');

var _themes = require('../utils/themes');

var _events = require('../utils/events');

var _ = require('.');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

// internal utility functions


// import constants


var BubbleBase = function (_Component) {
  _inherits(BubbleBase, _Component);

  // define static properties
  function BubbleBase(props) {
    _classCallCheck(this, BubbleBase);

    // define ref
    var _this = _possibleConstructorReturn(this, (BubbleBase.__proto__ || Object.getPrototypeOf(BubbleBase)).call(this, props));

    _this._hasEventListeners = false;

    _this._handleScrollEventListener = function (action) {
      // const rootNode = this.rootElement;
      var rootElement = _this.rootElement;

      if (rootElement) {
        var scrollableNode = _this._getFirstScrollableParent(rootElement);
        if (scrollableNode) {
          if (action === 'add') {
            scrollableNode.addEventListener('scroll', _this._updatePosition);
          } else if (action === 'remove') {
            scrollableNode.removeEventListener('scroll', _this._updatePosition);
          }
        }
      }
    };

    _this._getFirstScrollableParent = function (element) {
      if (element == null) return null;
      var rootElement = _this.rootElement;

      var node = {}.hasOwnProperty.call(element, 'current') ? element.current : element;

      if (rootElement) {
        if (node === rootElement.current || node.scrollHeight <= node.clientHeight) {
          return _this._getFirstScrollableParent(node.parentElement);
        }
      }

      return node;
    };

    _this._updatePosition = function () {
      var _this$props = _this.props,
          isOpeningUpward = _this$props.isOpeningUpward,
          targetRef = _this$props.targetRef;
      var rootElement = _this.rootElement;


      var target = targetRef && typeof targetRef !== 'string' ? targetRef.current : null;

      // Without a target, try to fallback to the parent node
      if (!target) {
        //  Only proceed if the root element is defined
        if (!rootElement || !rootElement.current) return;
        target = rootElement.current.parentElement;
      }

      var targetRect = target.getBoundingClientRect();

      var positionY = void 0;
      if (isOpeningUpward) {
        // Since we don't know the height of the bubble before rendering it we positioning
        // it with { bottom: XYpx } (within the viewport) and need this calculation:
        positionY = window.innerHeight - targetRect.top + 20;
      } else {
        positionY = targetRect.bottom + 20;
      }

      var position = {
        width: targetRect.width,
        positionX: targetRect.left,
        positionY: positionY
      };
      _this.setState({ position: position });
    };

    _this.rootElement = (0, _createRef2.default)();

    var context = props.context,
        themeId = props.themeId,
        theme = props.theme,
        themeOverrides = props.themeOverrides;


    _this.state = {
      composedTheme: (0, _themes.composeTheme)((0, _themes.addThemeId)(theme || context.theme, themeId), (0, _themes.addThemeId)(themeOverrides, themeId), context.ROOT_THEME_API),
      position: null
    };
    return _this;
  }
  // declare ref types


  _createClass(BubbleBase, [{
    key: 'componentDidMount',
    value: function componentDidMount() {
      var _this2 = this;

      setTimeout(function () {
        if (_this2.props.isFloating) _this2._updatePosition();
      }, 0);
    }
  }, {
    key: 'componentWillReceiveProps',
    value: function componentWillReceiveProps(nextProps) {
      (0, _themes.didThemePropsChange)(this.props, nextProps, this.setState.bind(this));
    }
  }, {
    key: 'componentWillUpdate',
    value: function componentWillUpdate(nextProps) {
      var isFloating = this.props.isFloating;
      // Add listeners when the bubble

      if (isFloating && !nextProps.isHidden && !this._hasEventListeners) {
        this._handleScrollEventListener('add');
        (0, _events.addDocumentListeners)(this._getDocumentEvents());
        window.addEventListener('resize', this._updatePosition);
        this._hasEventListeners = true;
      }
    }
  }, {
    key: 'componentDidUpdate',
    value: function componentDidUpdate(prevProps) {
      var isHidden = this.props.isHidden;

      var didVisibilityChange = isHidden !== prevProps.isHidden;
      var wasBubbleHidden = !prevProps.isHidden && isHidden;

      if (wasBubbleHidden) this._removeAllEventListeners();
      if (didVisibilityChange) this._updatePosition();
    }
  }, {
    key: 'componentWillUnmount',
    value: function componentWillUnmount() {
      if (this._hasEventListeners) this._removeAllEventListeners();
    }

    // =========== PRIVATE HELPERS ==============

  }, {
    key: '_removeAllEventListeners',
    value: function _removeAllEventListeners() {
      if (this._hasEventListeners) {
        (0, _events.removeDocumentListeners)(this._getDocumentEvents());
        this._handleScrollEventListener('remove');
        window.removeEventListener('resize', this._updatePosition);
        this._hasEventListeners = false;
      }
    }
  }, {
    key: '_getDocumentEvents',
    value: function _getDocumentEvents() {
      return {
        resize: this._updatePosition,
        scroll: this._updatePosition
      };
    }
  }, {
    key: 'render',
    value: function render() {
      // destructuring props ensures only the "...rest" get passed down
      var _props = this.props,
          skin = _props.skin,
          theme = _props.theme,
          themeOverrides = _props.themeOverrides,
          context = _props.context,
          rest = _objectWithoutProperties(_props, ['skin', 'theme', 'themeOverrides', 'context']);

      var BubbleSkin = skin || context.skins[_.IDENTIFIERS.BUBBLE];

      return _react2.default.createElement(BubbleSkin, _extends({
        rootRef: this.rootElement,
        position: this.state.position,
        theme: this.state.composedTheme
      }, rest));
    }
  }]);

  return BubbleBase;
}(_react.Component);

BubbleBase.displayName = 'Bubble';
BubbleBase.defaultProps = {
  context: (0, _withTheme.createEmptyContext)(),
  isHidden: false,
  isFloating: false,
  isOpeningUpward: false,
  isTransparent: true,
  arrowRelativeToTip: false,
  theme: null,
  themeId: _.IDENTIFIERS.BUBBLE,
  themeOverrides: {}
};
var Bubble = exports.Bubble = (0, _withTheme.withTheme)(BubbleBase);
//# sourceMappingURL=Bubble.js.map