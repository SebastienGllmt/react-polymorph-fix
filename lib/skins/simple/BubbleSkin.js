'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.BubbleSkin = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _props = require('../../utils/props');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

// external libraries


// internal utility functions


var BubbleSkin = exports.BubbleSkin = function BubbleSkin(props) {
  var _ref;

  var arrowRelativeToTip = props.arrowRelativeToTip,
      theme = props.theme,
      themeId = props.themeId;


  return _react2.default.createElement(
    'div',
    _extends({
      ref: props.rootRef
    }, (0, _props.pickDOMProps)(props), {
      className: (0, _classnames2.default)([props.className, theme[themeId].root, props.isOpeningUpward ? theme[themeId].openUpward : null, props.isTransparent ? theme[themeId].transparent : null, props.isFloating ? theme[themeId].isFloating : null, props.isHidden ? theme[themeId].isHidden : null]),
      style: props.position && (_ref = {}, _defineProperty(_ref, props.isOpeningUpward ? 'bottom' : 'top', props.position.positionY), _defineProperty(_ref, 'left', props.position.positionX), _defineProperty(_ref, 'width', props.position.width), _ref)
    }),
    _react2.default.createElement(
      'div',
      { className: theme[themeId].bubble, 'data-bubble-container': true },
      props.children,
      arrowRelativeToTip && _react2.default.createElement('span', { className: theme[themeId].arrow, 'data-bubble-arrow': true })
    ),
    !arrowRelativeToTip && _react2.default.createElement('span', { className: theme[themeId].arrow, 'data-bubble-arrow': true })
  );
};
//# sourceMappingURL=BubbleSkin.js.map