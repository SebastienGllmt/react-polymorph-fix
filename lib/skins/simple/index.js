'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SimpleSkins = undefined;

var _SimpleSkins;

var _components = require('../../components');

var _AutocompleteSkin = require('./AutocompleteSkin');

var _BubbleSkin = require('./BubbleSkin');

var _ButtonSkin = require('./ButtonSkin');

var _CheckboxSkin = require('./CheckboxSkin');

var _FormFieldSkin = require('./FormFieldSkin');

var _HeaderSkin = require('./HeaderSkin');

var _InfiniteScrollSkin = require('./InfiniteScrollSkin');

var _InputSkin = require('./InputSkin');

var _LoadingSpinnerSkin = require('./LoadingSpinnerSkin');

var _ModalSkin = require('./ModalSkin');

var _OptionsSkin = require('./OptionsSkin');

var _ProgressBarSkin = require('./ProgressBarSkin');

var _RadioSkin = require('./RadioSkin');

var _SelectSkin = require('./SelectSkin');

var _SwitchSkin = require('./SwitchSkin');

var _TextAreaSkin = require('./TextAreaSkin');

var _TogglerSkin = require('./TogglerSkin');

var _TooltipSkin = require('./TooltipSkin');

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var SimpleSkins = exports.SimpleSkins = (_SimpleSkins = {}, _defineProperty(_SimpleSkins, _components.IDENTIFIERS.AUTOCOMPLETE, _AutocompleteSkin.AutocompleteSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.BUBBLE, _BubbleSkin.BubbleSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.BUTTON, _ButtonSkin.ButtonSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.CHECKBOX, _CheckboxSkin.CheckboxSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.FORM_FIELD, _FormFieldSkin.FormFieldSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.HEADER, _HeaderSkin.HeaderSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.INFINITE_SCROLL, _InfiniteScrollSkin.InfiniteScrollSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.INPUT, _InputSkin.InputSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.LOADING_SPINNER, _LoadingSpinnerSkin.LoadingSpinnerSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.MODAL, _ModalSkin.ModalSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.OPTIONS, _OptionsSkin.OptionsSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.PROGRESS_BAR, _ProgressBarSkin.ProgressBarSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.RADIO, _RadioSkin.RadioSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.SELECT, _SelectSkin.SelectSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.SWITCH, _SwitchSkin.SwitchSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.TEXT_AREA, _TextAreaSkin.TextAreaSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.TOGGLER, _TogglerSkin.TogglerSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.TOOLTIP, _TooltipSkin.TooltipSkin), _SimpleSkins);
//# sourceMappingURL=index.js.map