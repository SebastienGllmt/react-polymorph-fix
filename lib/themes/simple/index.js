'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SimpleTheme = undefined;

var _SimpleTheme;

var _components = require('../../components');

var _SimpleAutocomplete = require('./SimpleAutocomplete.scss');

var _SimpleAutocomplete2 = _interopRequireDefault(_SimpleAutocomplete);

var _SimpleBubble = require('./SimpleBubble.scss');

var _SimpleBubble2 = _interopRequireDefault(_SimpleBubble);

var _SimpleButton = require('./SimpleButton.scss');

var _SimpleButton2 = _interopRequireDefault(_SimpleButton);

var _SimpleCheckbox = require('./SimpleCheckbox.scss');

var _SimpleCheckbox2 = _interopRequireDefault(_SimpleCheckbox);

var _SimpleFlex = require('./SimpleFlex.scss');

var _SimpleFlex2 = _interopRequireDefault(_SimpleFlex);

var _SimpleFormField = require('./SimpleFormField.scss');

var _SimpleFormField2 = _interopRequireDefault(_SimpleFormField);

var _SimpleGrid = require('./SimpleGrid.scss');

var _SimpleGrid2 = _interopRequireDefault(_SimpleGrid);

var _SimpleGutter = require('./SimpleGutter.scss');

var _SimpleGutter2 = _interopRequireDefault(_SimpleGutter);

var _SimpleHeader = require('./SimpleHeader.scss');

var _SimpleHeader2 = _interopRequireDefault(_SimpleHeader);

var _SimpleInfiniteScroll = require('./SimpleInfiniteScroll.scss');

var _SimpleInfiniteScroll2 = _interopRequireDefault(_SimpleInfiniteScroll);

var _SimpleInput = require('./SimpleInput.scss');

var _SimpleInput2 = _interopRequireDefault(_SimpleInput);

var _SimpleLoadingSpinner = require('./SimpleLoadingSpinner.scss');

var _SimpleLoadingSpinner2 = _interopRequireDefault(_SimpleLoadingSpinner);

var _SimpleModal = require('./SimpleModal.scss');

var _SimpleModal2 = _interopRequireDefault(_SimpleModal);

var _SimpleOptions = require('./SimpleOptions.scss');

var _SimpleOptions2 = _interopRequireDefault(_SimpleOptions);

var _SimpleProgressBar = require('./SimpleProgressBar.scss');

var _SimpleProgressBar2 = _interopRequireDefault(_SimpleProgressBar);

var _SimpleRadio = require('./SimpleRadio.scss');

var _SimpleRadio2 = _interopRequireDefault(_SimpleRadio);

var _SimpleSelect = require('./SimpleSelect.scss');

var _SimpleSelect2 = _interopRequireDefault(_SimpleSelect);

var _SimpleSwitch = require('./SimpleSwitch.scss');

var _SimpleSwitch2 = _interopRequireDefault(_SimpleSwitch);

var _SimpleTextArea = require('./SimpleTextArea.scss');

var _SimpleTextArea2 = _interopRequireDefault(_SimpleTextArea);

var _SimpleToggler = require('./SimpleToggler.scss');

var _SimpleToggler2 = _interopRequireDefault(_SimpleToggler);

var _SimpleTooltip = require('./SimpleTooltip.scss');

var _SimpleTooltip2 = _interopRequireDefault(_SimpleTooltip);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
// import theme IDENTIFIERS constants


// css modules plugin converts all imports below into plain objects


// SimpleTheme is a plain object serving as the default export.
// Each key is named after a component and each key's value
// is the component's corresponding theme. The user can
// pass this entire obj directly to ThemeProvider via the "theme" prop
var SimpleTheme = exports.SimpleTheme = (_SimpleTheme = {}, _defineProperty(_SimpleTheme, _components.IDENTIFIERS.AUTOCOMPLETE, _SimpleAutocomplete2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.BUBBLE, _SimpleBubble2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.BUTTON, _SimpleButton2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.CHECKBOX, _SimpleCheckbox2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.FLEX, _SimpleFlex2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.FORM_FIELD, _SimpleFormField2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.GRID, _SimpleGrid2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.GUTTER, _SimpleGutter2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.HEADER, _SimpleHeader2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.INFINITE_SCROLL, _SimpleInfiniteScroll2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.INPUT, _SimpleInput2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.LOADING_SPINNER, _SimpleLoadingSpinner2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.MODAL, _SimpleModal2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.OPTIONS, _SimpleOptions2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.PROGRESS_BAR, _SimpleProgressBar2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.RADIO, _SimpleRadio2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.SELECT, _SimpleSelect2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.SWITCH, _SimpleSwitch2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.TEXT_AREA, _SimpleTextArea2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.TOGGLER, _SimpleToggler2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.TOOLTIP, _SimpleTooltip2.default), _SimpleTheme);
//# sourceMappingURL=index.js.map