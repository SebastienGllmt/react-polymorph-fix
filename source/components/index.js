// @flow
export const IDENTIFIERS = {
  AUTOCOMPLETE: 'autocomplete',
  BUBBLE: 'bubble',
  BUTTON: 'button',
  CHECKBOX: 'checkbox',
  FLEX: 'flex',
  FORM_FIELD: 'formfield',
  GRID: 'grid',
  GUTTER: 'gutter',
  HEADER: 'header',
  INFINITE_SCROLL: 'infinitescroll',
  INPUT: 'input',
  LOADING_SPINNER: 'loadingspinner',
  MODAL: 'modal',
  OPTIONS: 'options',
  PROGRESS_BAR: 'progressbar',
  RADIO: 'radio',
  SELECT: 'select',
  SWITCH: 'switch',
  TEXT_AREA: 'textarea',
  TOGGLER: 'toggler',
  TOOLTIP: 'tooltip'
};
