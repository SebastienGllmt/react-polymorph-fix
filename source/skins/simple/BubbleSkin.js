// @flow
import React from 'react';
import type { Node, ElementRef } from 'react';

// external libraries
import classnames from 'classnames';

// internal utility functions
import { pickDOMProps } from '../../utils/props';

type Props = {
  children?: ?Node,
  className: string,
  isFloating: boolean,
  isHidden: boolean,
  isOpeningUpward: boolean,
  isTransparent: boolean,
  arrowRelativeToTip: boolean,
  position: Object,
  rootRef: ElementRef<*>,
  theme: Object,
  themeId: string
};

export const BubbleSkin = (props: Props) => {
  const { arrowRelativeToTip, theme, themeId } = props;

  return (
    <div
      ref={props.rootRef}
      {...pickDOMProps(props)}
      className={classnames([
        props.className,
        theme[themeId].root,
        props.isOpeningUpward ? theme[themeId].openUpward : null,
        props.isTransparent ? theme[themeId].transparent : null,
        props.isFloating ? theme[themeId].isFloating : null,
        props.isHidden ? theme[themeId].isHidden : null
      ])}
      style={
        props.position && {
          [props.isOpeningUpward ? 'bottom' : 'top']: props.position.positionY,
          left: props.position.positionX,
          width: props.position.width
        }
      }
    >
      <div className={theme[themeId].bubble} data-bubble-container>
        {props.children}
        {arrowRelativeToTip && <span className={theme[themeId].arrow} data-bubble-arrow />}
      </div>
      {!arrowRelativeToTip && <span className={theme[themeId].arrow} data-bubble-arrow />}
    </div>
  );
};
