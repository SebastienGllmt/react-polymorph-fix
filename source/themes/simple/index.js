// @flow
// import theme IDENTIFIERS constants
import { IDENTIFIERS } from '../../components';

// css modules plugin converts all imports below into plain objects
import SimpleAutocomplete from './SimpleAutocomplete.scss';
import SimpleBubble from './SimpleBubble.scss';
import SimpleButton from './SimpleButton.scss';
import SimpleCheckbox from './SimpleCheckbox.scss';
import SimpleFlex from './SimpleFlex.scss';
import SimpleFormField from './SimpleFormField.scss';
import SimpleGrid from './SimpleGrid.scss';
import SimpleGutter from './SimpleGutter.scss';
import SimpleHeader from './SimpleHeader.scss';
import SimpleInfiniteScroll from './SimpleInfiniteScroll.scss';
import SimpleInput from './SimpleInput.scss';
import SimpleLoadingSpinner from './SimpleLoadingSpinner.scss';
import SimpleModal from './SimpleModal.scss';
import SimpleOptions from './SimpleOptions.scss';
import SimpleProgressBar from './SimpleProgressBar.scss';
import SimpleRadio from './SimpleRadio.scss';
import SimpleSelect from './SimpleSelect.scss';
import SimpleSwitch from './SimpleSwitch.scss';
import SimpleTextArea from './SimpleTextArea.scss';
import SimpleToggler from './SimpleToggler.scss';
import SimpleTooltip from './SimpleTooltip.scss';

// SimpleTheme is a plain object serving as the default export.
// Each key is named after a component and each key's value
// is the component's corresponding theme. The user can
// pass this entire obj directly to ThemeProvider via the "theme" prop
export const SimpleTheme = {
  [IDENTIFIERS.AUTOCOMPLETE]: SimpleAutocomplete,
  [IDENTIFIERS.BUBBLE]: SimpleBubble,
  [IDENTIFIERS.BUTTON]: SimpleButton,
  [IDENTIFIERS.CHECKBOX]: SimpleCheckbox,
  [IDENTIFIERS.FLEX]: SimpleFlex,
  [IDENTIFIERS.FORM_FIELD]: SimpleFormField,
  [IDENTIFIERS.GRID]: SimpleGrid,
  [IDENTIFIERS.GUTTER]: SimpleGutter,
  [IDENTIFIERS.HEADER]: SimpleHeader,
  [IDENTIFIERS.INFINITE_SCROLL]: SimpleInfiniteScroll,
  [IDENTIFIERS.INPUT]: SimpleInput,
  [IDENTIFIERS.LOADING_SPINNER]: SimpleLoadingSpinner,
  [IDENTIFIERS.MODAL]: SimpleModal,
  [IDENTIFIERS.OPTIONS]: SimpleOptions,
  [IDENTIFIERS.PROGRESS_BAR]: SimpleProgressBar,
  [IDENTIFIERS.RADIO]: SimpleRadio,
  [IDENTIFIERS.SELECT]: SimpleSelect,
  [IDENTIFIERS.SWITCH]: SimpleSwitch,
  [IDENTIFIERS.TEXT_AREA]: SimpleTextArea,
  [IDENTIFIERS.TOGGLER]: SimpleToggler,
  [IDENTIFIERS.TOOLTIP]: SimpleTooltip
};
