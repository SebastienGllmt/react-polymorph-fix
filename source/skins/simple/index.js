// @flow
import { IDENTIFIERS } from '../../components';
import { AutocompleteSkin } from './AutocompleteSkin';
import { BubbleSkin } from './BubbleSkin';
import { ButtonSkin } from './ButtonSkin';
import { CheckboxSkin } from './CheckboxSkin';
import { FormFieldSkin } from './FormFieldSkin';
import { HeaderSkin } from './HeaderSkin';
import { InfiniteScrollSkin } from './InfiniteScrollSkin';
import { InputSkin } from './InputSkin';
import { LoadingSpinnerSkin } from './LoadingSpinnerSkin';
import { ModalSkin } from './ModalSkin';
import { OptionsSkin } from './OptionsSkin';
import { ProgressBarSkin } from './ProgressBarSkin';
import { RadioSkin } from './RadioSkin';
import { SelectSkin } from './SelectSkin';
import { SwitchSkin } from './SwitchSkin';
import { TextAreaSkin } from './TextAreaSkin';
import { TogglerSkin } from './TogglerSkin';
import { TooltipSkin } from './TooltipSkin';

export const SimpleSkins = {
  [IDENTIFIERS.AUTOCOMPLETE]: AutocompleteSkin,
  [IDENTIFIERS.BUBBLE]: BubbleSkin,
  [IDENTIFIERS.BUTTON]: ButtonSkin,
  [IDENTIFIERS.CHECKBOX]: CheckboxSkin,
  [IDENTIFIERS.FORM_FIELD]: FormFieldSkin,
  [IDENTIFIERS.HEADER]: HeaderSkin,
  [IDENTIFIERS.INFINITE_SCROLL]: InfiniteScrollSkin,
  [IDENTIFIERS.INPUT]: InputSkin,
  [IDENTIFIERS.LOADING_SPINNER]: LoadingSpinnerSkin,
  [IDENTIFIERS.MODAL]: ModalSkin,
  [IDENTIFIERS.OPTIONS]: OptionsSkin,
  [IDENTIFIERS.PROGRESS_BAR]: ProgressBarSkin,
  [IDENTIFIERS.RADIO]: RadioSkin,
  [IDENTIFIERS.SELECT]: SelectSkin,
  [IDENTIFIERS.SWITCH]: SwitchSkin,
  [IDENTIFIERS.TEXT_AREA]: TextAreaSkin,
  [IDENTIFIERS.TOGGLER]: TogglerSkin,
  [IDENTIFIERS.TOOLTIP]: TooltipSkin
};
