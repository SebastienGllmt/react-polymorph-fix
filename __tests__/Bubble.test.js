import React from 'react';

import { Bubble } from '../source/components/Bubble';
import { BubbleSkin } from '../source/skins/simple/BubbleSkin';
import { renderInSimpleTheme } from './helpers/theming';

test('Bubble renders correctly', () => {
  const component = renderInSimpleTheme(
    <Bubble skin={BubbleSkin} />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Bubble renders isOpeningUpward={true}', () => {
  const component = renderInSimpleTheme(
    <Bubble isOpeningUpward skin={BubbleSkin} />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Bubble renders isTransparent={false}', () => {
  const component = renderInSimpleTheme(
    <Bubble isTransparent={false} skin={BubbleSkin} />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Bubble renders isHidden={true}', () => {
  const component = renderInSimpleTheme(
    <Bubble isHidden skin={BubbleSkin} />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Bubble renders isFloating={true}', () => {
  const component = renderInSimpleTheme(
    <Bubble isFloating skin={BubbleSkin} />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
