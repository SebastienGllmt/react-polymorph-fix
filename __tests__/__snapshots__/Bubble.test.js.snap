// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bubble renders correctly 1`] = `
<div
  className="root transparent"
  style={null}
>
  <div
    className="bubble"
    data-bubble-container={true}
  />
  <span
    className="arrow"
    data-bubble-arrow={true}
  />
</div>
`;

exports[`Bubble renders isFloating={true} 1`] = `
<div
  className="root transparent isFloating"
  style={null}
>
  <div
    className="bubble"
    data-bubble-container={true}
  />
  <span
    className="arrow"
    data-bubble-arrow={true}
  />
</div>
`;

exports[`Bubble renders isHidden={true} 1`] = `
<div
  className="root transparent isHidden"
  style={null}
>
  <div
    className="bubble"
    data-bubble-container={true}
  />
  <span
    className="arrow"
    data-bubble-arrow={true}
  />
</div>
`;

exports[`Bubble renders isOpeningUpward={true} 1`] = `
<div
  className="root openUpward transparent"
  style={null}
>
  <div
    className="bubble"
    data-bubble-container={true}
  />
  <span
    className="arrow"
    data-bubble-arrow={true}
  />
</div>
`;

exports[`Bubble renders isTransparent={false} 1`] = `
<div
  className="root"
  style={null}
>
  <div
    className="bubble"
    data-bubble-container={true}
  />
  <span
    className="arrow"
    data-bubble-arrow={true}
  />
</div>
`;
